.azure/linux-cabal.yml

```diff
@@ -0,0 +1,75 @@
+jobs:
+- job: Linux_Cabal
+  timeoutInMinutes: 0
+  pool:
+    vmImage: ubuntu-16.04
+  strategy:
+    matrix:
+      ghc-8.6.5:
+        GHC_VERSION: "8.6.5"
+      ghc-8.4.4:
+        GHC_VERSION: "8.4.4"
+  variables:
+    CABAL_VERSION: "3.0"
+  steps:
+  - task: Cache@2
+    inputs:
+      key: '"cabal-store" | "$(Agent.OS)" | "$(CABAL_VERSION)" | "$(GHC_VERSION)" | $(Build.SourcesDirectory)/cabal.project | $(Build.SourcesDirectory)/haskell-ide-engine.cabal | $(Build.SourcesDirectory)/hie-plugin-api/hie-plugin-api.cabal'
+      path: .azure-cache
+      cacheHitVar: CACHE_RESTORED
+    displayName: "Download cache"
+  - bash: |
+      mkdir -p $HOME/.cabal
+      tar -vxzf .azure-cache/cabal-root.tar.gz -C /
+      mkdir -p $HOME/.ghc
+      tar -vxzf .azure-cache/ghc-root.tar.gz -C /
+      mkdir -p dist-newtyle
+      tar -vxzf .azure-cache/cabal-dist.tar.gz
+    displayName: "Unpack cache"
+    condition: eq(variables.CACHE_RESTORED, 'true')
+  - bash: |
+      git submodule sync
+      git submodule update --init
+    displayName: Sync submodules
+  - bash: |
+      source .azure/linux-cabal.bashrc
+      cabal v2-update
+      cabal v2-build --only-dependencies
+    displayName: Build dependencies
+  - bash: |
+      source .azure/linux-cabal.bashrc
+      cabal v2-build
+    displayName: Build `hie`
+  - bash: |
+      source .azure/linux-cabal.bashrc
+      cabal v2-install # `hie` binary required locally for tests
+    displayName: Install `hie`
+  - bash: |
+      source .azure/linux-cabal.bashrc
+      cabal v2-build --enable-tests --enable-benchmarks --only-dependencies
+    displayName: Build Test-dependencies
+  - bash: |
+      sudo apt update
+      sudo apt install z3
+    displayName: "Install Runtime Test-Dependencies: z3"
+  - bash: |
+      source .azure/linux-cabal.bashrc
+      # to not resintall hie
+      cd $(Agent.TempDirectory)
+      cabal v2-install liquidhaskell-0.8.6.2 -w /opt/ghc/8.6.5/bin/ghc
+    displayName: "Install Runtime Test-Dependencies: liquidhaskell"
+  - bash: |
+      source .azure/linux-cabal.bashrc
+      cabal v2-build hoogle
+      cabal v2-exec hoogle generate
+    displayName: "Install Runtime Test-Dependencies: hoogle database"
+  - bash: |
+      source .azure/linux-cabal.bashrc
+      cabal v2-test -j1
```

Comment:
why only one job?

Reply:
Sadly it fails if testing use parallelism, like in circleci:
https://github.com/haskell/haskell-ide-engine/blob/92ac896029a5ba5f740ebca168533c83848548c6/.circleci/config.yml#L85

Maybe we could investigate why...

Reply:
A TODO maybe for the future?